Rename all Layer and Source properties 'projection' by 'crs'.

gchoqueux opened the pull request:
## Description
Rename all **`Layer`** and **`Source`** properties `projection` by **`crs`**.

**Deprecated**: **`Source.projection`** and **`Layer.projection`** properties are deprecated.
- Use **`crs`** instead of **`projection`**.
## Motivation
Have same **`crs`** projection property for : **`Extent, Coordinates, Features, Layer and Source`**

gchoqueux committed:
fix(test): update deprecated 'projection' parameter/property to 'crs'.

gchoqueux committed:
fix(examples): update deprecated 'projection' parameter/property to 'crs'.

zarov commented on src/Layer/Layer.js:
```diff
@@ -75,6 +75,11 @@ class Layer extends THREE.EventDispatcher {
     constructor(id, config = {}) {
+        /* istanbul ignore next */
+        if (config.projection) {
+            console.warn('Layer projection parameter is deprecated, use crs instead of.');
```
`use crs instead of` -> `use crs instead`.

gchoqueux replied:
thx, done

zarov commented on src/Layer/OrientedImageLayer.js:
```diff
@@ -93,14 +93,19 @@ class OrientedImageLayer extends GeometryLayer {
+        /* istanbul ignore next */
+        if (config.projection) {
+            console.warn('OrientedImageLayer projection parameter is deprecated, use crs instead of.');
+            config.crs = config.cr || config.projection;
```
Typo: `cr` -> `crs`

gchoqueux replied:
done

zarov commented on src/Source/FileSource.js:
```diff
@@ -100,19 +100,23 @@ const ext = new Extent('EPSG:4326', [0, 0, 0, 0]);
-     * @param {string} crsOut - The projection of the output data after parsing.
+     * FileSource and {@link Source}. Only `crs` is mandatory, but if it
+     * presents in `parsedData` under the property `crs`, it is fine.
+     * @param {string} crsOut - The crs projection of the output data after parsing.
```
Where is this parameter ?

gchoqueux replied:
removed

zarov commented on src/Source/WFSSource.js:
```diff
@@ -92,18 +92,22 @@ import CRS from 'Core/Geographic/Crs';
     constructor(source) {
+        if (source.projection) {
+            console.warn('WFSSource projection parameter is deprecated, use crs instead of.');
+            source.crs = source.cr || source.projection;
```
`cr` -> `crs`

gchoqueux replied:
done

zarov reviewed (approved):
You missed some in this [tutorial](https://github.com/iTowns/itowns/blob/master/docs/tutorials/Display-a-geometry-layer.md)

gchoqueux committed:
refactor(Core): rename all properties 'projection' by 'crs'.

Deprecated: Source.projection and Layer.projection properties are deprecated.
use 'crs' instead of 'projection'.

gchoqueux committed:
fix(doc): update deprecated 'projection' parameter/property to 'crs'.

gchoqueux commented:
> You missed some in this tutorial

done, thx
